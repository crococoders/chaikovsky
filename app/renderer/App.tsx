var React = require("react");

// Usage exmaple of webpack alias (from "./components")
import { Heading, Example } from "@components";

import { hot as App } from "react-hot-loader/root";

export default App(
  (): JSX.Element => (
    <div>
      <h1>Your Electron Project</h1>
      <Heading title="Hello" />
      <Example>Element</Example>
    </div>
  ),
);
