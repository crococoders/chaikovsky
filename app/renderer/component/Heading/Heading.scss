.heading-title {
  @include title($size: 50px);
  color: blue;
}
