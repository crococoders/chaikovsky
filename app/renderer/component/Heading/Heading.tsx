import * as React from 'react';

type HeadingProps = {
  title?: string;
  children?: React.ReactNode;
};

type HeadingState = {
  counter: number;
};

export class Heading extends React.Component<HeadingProps, HeadingState> {
  state: HeadingState = {
    counter: 0,
  };

  handleClick = () => {
    this.setState({ counter: this.state.counter + 1 });
  };

  render() {
    return (
      <h1 onClick={this.handleClick}>
        {this.props.title ? this.props.title : this.props.children}
      </h1>
    );
  }
}
