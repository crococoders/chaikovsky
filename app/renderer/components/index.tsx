import { Heading } from "./Heading/Heading";
import { Example } from "./Example/Example";
import { Slider } from "./Slider/Slider";

export { Heading, Example, Slider };
