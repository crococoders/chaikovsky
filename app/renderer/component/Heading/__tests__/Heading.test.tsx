import "jsdom-global/register";

import React from "react";
import { shallow } from "enzyme";
import { Heading } from "../Heading";

import renderer from "react-test-renderer";

describe("Unit testing Heading", () => {
  it("Should use children", () => {
    const wrapper = shallow(<Heading>Hello Jest!</Heading>);
    expect(wrapper.html()).toMatch('<h1 class="heading-title">Hello Jest!</h1>');
  });

  it("Should use props", () => {
    const wrapper = shallow(<Heading title="Hello Jest!" />);
    expect(wrapper.html()).toMatch('<h1 class="heading-title">Hello Jest!</h1>');
  });

  it("Should check if state is 1", () => {
    const wrapper = shallow(<Heading title="Hello Jest!" />);
    wrapper.find("h1").simulate("click");
    expect(wrapper.state("counter")).toEqual(1);
  });
});

describe("Snapshot testing Heading", () => {
  it("Should renders correctly", () => {
    const tree = renderer.create(<Heading title="Hello Jest1!" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
