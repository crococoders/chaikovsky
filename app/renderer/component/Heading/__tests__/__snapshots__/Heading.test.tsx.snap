// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Heading Should renders correctly 1`] = `
<h1
  onClick={[Function]}
>
  Hello Jest1!
</h1>
`;
